File: packages/paysafe-card-payments/ios/PSCardholderNameViewManager.swift
//
//  PSCardholderNameViewManager.swift
//
//  Copyright © 2025 Paysafe. All rights reserved.
//

import PaysafePaymentsSDK
import React
import UIKit

@objc(PSCardholderNameView) class PSCardholderNameViewManager: RCTViewManager {

  override func view() -> UIView? {
    PSCardholderNameInputView()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

File: packages/paysafe-card-payments/android/src/main/java/com/paysafecardpayments/holderName/PSCardholderNameViewManager.kt
// Copyright Paysafe 2025. All rights reserved.

package com.paysafecardpayments.holderName

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext

class PSCardholderNameViewManager(
  private val reactContext: ReactApplicationContext,
  private val viewFactory: (android.app.Activity) -> PSCardholderNameWrapperView = { PSCardholderNameWrapperView(it) }
) : SimpleViewManager<PSCardholderNameWrapperView>() {

  override fun getName(): String = "PSCardholderNameView"

  override fun createViewInstance(reactContext: ThemedReactContext): PSCardholderNameWrapperView {
    return createWrapperView(reactContext)
  }

  internal fun createWrapperView(reactContext: ThemedReactContext): PSCardholderNameWrapperView {
    val activity = reactContext.currentActivity
      ?: throw IllegalStateException("Activity is null when creating PSCardholderNameWrapperView")
    return viewFactory(activity)
  }
}
